### DISCUSSION TITLE
overlay-buster.js

### DISCUSSION THREAD

**Discussion by ghajini** [uBlockOrigin/uBlock-issues#3470] [General] [answered]
when i use `stream.hownetwork.xyz##+js(overlay-buster)`

it works at `https://stream.hownetwork.xyz/video.php?id=7df95eed8c6a0ad7fba6e47bad1b1189`

but not on `https://playeriframe.lol/iframe.php?url=https%3A%2F%2Fstream.hownetwork.xyz%2Fvideo.php%3Fid%3D7df95eed8c6a0ad7fba6e47bad1b1189`

is it by design?
@gorhill can you see?

it was for https://github.com/uBlockOrigin/uBOL-home/issues/250 since ad id changes from **###adyes** to **###thankyou**

**Comment by ghost** [comment, marked as answer]
Works in top frame only: https://github.com/gorhill/uBlock/blob/a94df7f3b27080ae2dcb3b914ace39c0c294d2f6/assets/resources/scriptlets.js#L638-L640

**Reply by gorhill** [reply]
I can look into changing this, this code hasn't been touched since a very long time.

**Reply by ghajini** [reply]
can this be worked

**Reply by gorhill** [reply]
Next dev build you can use `stream.hownetwork.xyz##+js(overlay-buster, true)` to allow execution for embedded contexts.

**Reply by ghost** [reply]
@gorhill typo, the `s` in `allFrames` is missing in `if` statement condition.

https://github.com/gorhill/uBlock/commit/fc231998b9357b74c46450856535f948145b884a

**Reply by gorhill** [reply]
Oops sorry, not sure what happened there, I typically always mechanically hit F8 which is bound to ESlint for the current edited file.

**Comment by ghajini** [comment]
Not related to specific case issue here but to **overlay-defuser**
when i use `vinovo.to##+js(overlay-defuser)` on `https://vinovo.to/e/6726894vf7k3qn8` with **firefox android**
then
- when i click fullscreen button in video near right corner ,video does not go into fullscreen ,video does not play ,remain blank ,reverts back to portrait mode ,throwing below error in console
- 
**Exited fullscreen because fullscreen element was removed from document.**
![firefox_WXCnAyc9RJ](https://github.com/user-attachments/assets/0623235c-707c-433e-bbfb-4d0334d176d3)

related to script = `https://pastebin.com/J1gMXUfr`

**Reply by ghajini** [reply]
@gorhill @gwarser can you see?

**Reply by ghost** [reply]
This is how it works. Removes all top, full window elements. It should stop if you wait 30s. 

***

"overlay-buster"

**Reply by ghajini** [reply]
right
i 'll resort to https://github.com/uBlockOrigin/uAssets/commit/f529c4b2e110face8f5f7ad9edc52cbe6eaa9fd7
because multiple clicks are boring on mobile to start the video

**Comment by ghajini** [comment]
also another case `https://rm.freex2line.online/loadfree/?redirect=aHR0cHM6Ly9jaW1hbm93LmNjLyVkOSU4MSVkOSU4YSVkOSU4NCVkOSU4NS12ZW5vbS10aGUtbGFzdC1kYW5jZS0yMDI0LSVkOSU4NSVkOCVhYSVkOCViMSVkOCVhYyVkOSU4NS93YXRjaGluZw==`

`freex2line.online##+js(overlay-buster)` does not prevent popups , is this same as https://github.com/uBlockOrigin/uBlock-issues/discussions/3551#discussioncomment-12315932

**Reply by ghost** [reply]
It does prevent popup on desktop. Does not on mobile?

***

There can be a slight delay before the element is removed (I noticed this on vinovo above), so maybe if you tap it immediately after load it will open a popup.

**Reply by ghajini** [reply]
iam using opera browser/firefox desktop

**Reply by ghajini** [reply]
sorry to mention @gorhill, @gwarser again,
i can reproduce only in ubo dev 1.62.1b6 edge/firefox
it works correctly on ubo stable 1.62 edge/firefox android

**link** = `https://rm.freex2line.online/loadon/?link=aHR0cHM6Ly9jaW1hbm93LmNjLyVkOSU4NSVkOCViMyVkOSU4NCVkOCViMyVkOSU4NC0lZDglYTclZDklODQlZDglYjQlZDklODUlZDklOGElZDglYjMlZDklOGEtJWQ4JWE3JWQ5JTg0JWQ4JWFkJWQ5JTg0JWQ5JTgyJWQ4JWE5LTEtJWQ4JWE3JWQ5JTg0JWQ4JWE3JWQ5JTg4JWQ5JTg0JWQ5JThhL3dhdGNoaW5nLw==`

if you get error on above link try `https://cimanow.cc/selary/%d9%85%d8%b3%d9%84%d8%b3%d9%84-%d8%b3%d9%8a%d8%af-%d8%a7%d9%84%d9%86%d8%a7%d8%b3-%d8%ac1/` 
then click 
![image](https://github.com/user-attachments/assets/935a3faf-ae1a-414c-b6d2-b7f5e04a7846)

**Reply by ghost** [reply]
I'm perm-blocked by Cloudflare captcha. What is the issue?

**Reply by ghajini** [reply]
- if you see blocked message then close that tab then clear cookies/data then open new tab then go to `https://cimanow.cc/home/`
click one of the movie & click the link as shown in image in previous reply
https://github.com/uBlockOrigin/uBlock-issues/discussions/3470#discussioncomment-12462121 which will navigate you to `freex2line.online` domain

- add `freex2line.online#@#body >a[id][href][target^="_blank"]` temporarily

- **issue** = `freex2line.online##+js(overlay-buster)` does not work on ubo dev1.62.1b6, it works on ubo 1.62

**Reply by gorhill** [reply]
There was a typo in b6, fixed in rc0: https://github.com/gorhill/uBlock/commit/31d82c494b196a37a8f30765b465acbfa4871f34
